packages/client/hmi-client/src/model-representation/service.ts

```diff
@@ -145,3 +147,128 @@ export const getModelRenderer = (
+	const modelType = getModelType(model);
+	switch (modelType) {
+		case AMRSchemaNames.REGNET:
+			return model.model?.parameters.find((p) => p.id === parameterId);
+		case AMRSchemaNames.PETRINET:
+		case AMRSchemaNames.STOCKFLOW:
+		default:
+			return model.semantics?.ode?.parameters?.find((p) => p.id === parameterId) ?? null;
+	}
+}
+
+/**
+ * Returns the metadata for the specified semantic type and parameter ID.
+ * @param {Model} model - The model object.
+ * @param {string} semanticType - The semantic type.
+ * @param {string} parameterId - The parameter ID.
+ * @returns {any} - The metadata.
+ */
+export function getMetadata(model: Model, semanticType: string, parameterId: string) {
+	return model.metadata?.[semanticType]?.[parameterId];
+}
+
+/**
+ * Returns the model initials based on the model type.
+ * @param {Model} model - The model object.
+ * @returns {Initial[]} - The model initials.
+ */
+export function getModelInitials(model: Model): Initial[] {
+	const modelType = getModelType(model);
+	switch (modelType) {
+		case AMRSchemaNames.REGNET:
+			return model.model?.vertices ?? [];
+		case AMRSchemaNames.PETRINET:
+		case AMRSchemaNames.STOCKFLOW:
+		default:
+			return model.semantics?.ode?.initials ?? [];
+	}
+}
+
+/**
+ * Returns the model initial with the specified ID.
+ * @param {Model} model - The model object.
+ * @param {string} initialId - The ID of the initial.
+ * @returns {Initial | null} - The model initial or null if not found.
+ */
+export function getInitial(model: Model, initialId: string): Initial | null {
+	const modelType = getModelType(model);
+	switch (modelType) {
+		case AMRSchemaNames.REGNET:
+			return model.model?.vertices.find((i) => i.id === initialId);
+		case AMRSchemaNames.PETRINET:
+		case AMRSchemaNames.STOCKFLOW:
+		default:
+			return model.semantics?.ode?.initials?.find((i) => i.target === initialId) ?? null;
+	}
+}
+
+/**
+ * Returns the timeseries for the specified semantic ID.
+ * @param {Model} model - The model object.
+ * @param {string} semanticId - The semantic ID.
+ * @returns {any} - The timeseries.
+ */
+export function getTimeseries(model: Model, semanticId: string) {
+	return model.metadata?.timeseries?.[semanticId];
+}
+
+/**
+ * Updates the metadata for the specified semantic ID, semantic type, and metadata key.
+ * @param {Model} model - The model object.
+ * @param {string} semanticId - The semantic ID.
+ * @param {string} semanticType - The semantic type.
+ * @param {string} metadataKey - The metadata key.
+ * @param {any} value - The new value.
+ */
+export function updateMetadata(
+	model: Model,
+	semanticId: string,
+	semanticType: string,
+	metadataKey: string,
+	value: any
+) {
+	if (!model.metadata?.[semanticType]?.[semanticId]) {
+		model.metadata ??= {};
+		model.metadata[semanticType] ??= {};
+		model.metadata[semanticType][semanticId] ??= {};
+	}
+	model.metadata[semanticType][semanticId][metadataKey] = value;
+}
```

Comment:
This feels both too abstract, and overly-fitting to the timeseries scenario. There are no clear indications that semanticType > semanticId > metadataKey.

Also this makes it not possible to do `metadata['notes'] = "i am a note"`cases of simple key-value assignment

Reply:
https://unchartedsoftware.slack.com/archives/C05PETG2GTE/p1712952982343699?thread_ts=1712951482.874169&cid=C05PETG2GTE